src/embind/embind.js

```diff
@@ -626,7 +626,35 @@ var LibraryEmbind = {
 
             var str;
             if(stdStringIsUTF8) {
-                str = UTF8ToString(value + 4, length);
+                //ensure null termination at one-past-end byte if not present yet
```

Comment:
This should not be necessary - std::string in C++11 guarantees that data()[size()] is the null character.

Reply:
Thanks, I would really like to drop this, but I am very sure that embind specific inheritance unit tests will break if I try it. Some of the type names in the unit test were not converted correctly, e.g. when checking for the name of the base class. Maybe we can check if there is something wrong with the affected unit test(s)?

Reply:
Huh, okay. I can't imagine what's going on. Maybe add a comment that the code should not be necessary but the tests fail without it?